src/ipc/ipc-common.c

```diff
@@ -212,10 +225,24 @@ void ipc_msg_send(struct ipc_msg *msg, void *data, bool high_priority)
 			list_item_append(&msg->list, &ipc->msg_list);
 	}
 
+	schedule_ipc_worker();
+
 out:
 	k_spin_unlock(&ipc->lock, key);
 }
 
+#ifdef __ZEPHYR__
+static void ipc_work_handler(struct k_work *work)
+{
+	struct ipc *ipc = ipc_get();
+
+	ipc_send_queued_msg();
+
+	if (!list_is_empty(&ipc->msg_list))
```

Comment:
do we need logic similar to the present one to avoid scheduling IPCs when preparing to suspend?

Reply:
@lyakh Good question. ipc_send_queued_msg() won't send anything in we are going to D0. Not sure if rescheduling the task would block D0 sequence (if the flag is set, we've already made the decision to power off). I guess it wouldn't hurt.

Reply:
I pushed V2 that stops re-queuing if D3 request flag is set to keep the old behaviour.